app/models/sac_cas/phone_number.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+module SacCas::PhoneNumber
+  extend ActiveSupport::Concern
+
+  included do
+    after_create :check_data_quality
+    after_destroy :check_data_quality
+  end
+
+  private
+
+  def check_data_quality
+    # prevent running the check twice
+    return if People::DataQualityChecker.attributes_to_check_changed?(contactable)
+
+    People::DataQualityChecker.new(contactable.reload).check_data_quality
```

Comment:
Das reload ist heikel, da dabei in einer Update Action allenfalls Informationen verloren gehen. Die Methode muss ohne Reload funktionieren, wie in Person auch.

Reply:
> da dabei in einer Update Action allenfalls Informationen verloren gehen

wie können infos verloren gehen bei einem after_create/destroy callback? ich habe es mehrmals lokal getestet und bin zum entschluss gekommen, dass es leider nur mit reload geht. hättest du eine idee, wie es anders geht?

Reply:
Das contactable kann ja z.B. das Person Objekt sein, welches im Controller aktualisiert wird. Im after_destroy Callback der PhoneNumber ist nicht klar, ob die Person bereits gespeichert wurde oder nicht. Mit dem reload prüfst du dann die Datenqualität der Version der Person aus der Datenbank, welche allenfalls die Änderungen vom Controller noch nicht enthält. 

Hast du eine Spec welche failed, wenn kein `reload` erfolgt? Was funktioniert dabei nicht? Allenfalls muss dort versucht werden, dass contactable auf der PhoneNumber wirklich das gleiche Objekt (object_id) ist wie die Person. So oder so braucht es wohl noch Specs, welche das Hinzufügen sowie das Löschen einer Telefonnummer via PeopleController testen, ohne das dabei andere Attribute angepasst werden.

Ein anderer Ansatz könnte sein, via before_save und before_destroy auf der PhoneNumber bei der Person ein Flag/Attribut (z.B. phone_numbers_changed) zu setzen und dieses bei Person#check_data_quality ebenfalls zu prüfen, um den Checker zu starten. So viel ich weiss werden Telefonnummern nur via Person aktualisiert, dann könnte das reichen.

Reply:
ich reloade jetzt stattdessen die phone numbers. die infos gehen nicht verloren, trotz reload.